lib/puppet/type/package.rb

```diff
@@ -74,6 +74,11 @@ module Puppet
         value. On packaging systems that manage configuration files separately
         from "normal" system files, you can uninstall config files by
         specifying `purged` as the ensure value. This defaults to `installed`.
+
+        Version numbers must match the full version to install, including
+        release if the provider uses a release moniker. Ranges or semver
+        patterns are not accepted. For example, to install the rpm
```

Comment:
Should we note provider-specific exceptions in the type reference? Or couch that statement with something along the lines of "... unless the package provider explicitly allows it"?